plugin.picture.googlephotos/main.py

```diff
@@ -0,0 +1,395 @@
+import code
+import sys
+import requests
+from pathlib import Path
+from resources.lib import auth
+import resources.lib.config as config
+from urllib import parse
+import xbmc
+import xbmcaddon
+import xbmcvfs
+import xbmcgui
+import xbmcplugin
+import threading
+import datetime
+import traceback
+
+
+from resources.lib.auth import read_credentials, get_device_code
+# from resources.lib.ui.custom_filter_dialog import FilterDialog
+import resources.lib.utils as utils
+
+base_url = sys.argv[0]
+addon_handle = int(sys.argv[1])
+qs = sys.argv[2][1:]
+args = parse.parse_qs(qs)
+
+mode = args.get('mode')
+token = None  # Access token
+
+# Addon dir path
+__addon__ = xbmcaddon.Addon()
+# addon_path = __addon__.getAddonInfo('path')
+profile_path = xbmcvfs.translatePath(
+    __addon__.getAddonInfo('profile'))
+token_folder = Path(profile_path + config.token_folder)
+token_path = None
+if args.get('token_filename'):
+    token_path = Path(token_folder / args.get('token_filename')[0])
+xbmcplugin.setContent(addon_handle, 'images')
+
+
+def new_account():
+    xbmc.log('Executing new_account function', xbmc.LOGDEBUG)
+
+    # Open dialog
+    baseUrl = __addon__.getSettingString('baseUrl')
+    dialog_msg = __addon__.getLocalizedString(30400) + f' {baseUrl}:\n'
+    title = __addon__.getLocalizedString(30401)
+    load_msg = __addon__.getLocalizedString(30423)
+    login_dialog = xbmcgui.DialogProgress()
+    login_dialog.create(title, dialog_msg + load_msg)
+
+    # Get User Code from auth server
+    code_json = get_device_code()
+    init_time = datetime.datetime.utcnow()
+    min_time_to_refresh = (
+        100 / code_json["expires_in"]) * code_json['interval']
+    login_dialog.update(100, message=dialog_msg + code_json["userCode"])
+
+    # Update progress dialog indicating time left for complete
+    sleep_time = code_json['interval'] * 1000
+    time = 99
+    while True:
+        login_dialog.update(time)
+        if login_dialog.iscanceled():
+            break
+        xbmc.sleep(sleep_time)
+        status_code = auth.fetch_and_save_token(
+            code_json['deviceCode'], token_folder)
+        if status_code == 200:
+            xbmc.executebuiltin(
```

Comment:
Please use the `xbmcgui.Dialog().notification()` method, instead of `xbmc.executebuiltin()`

Reply:
Done